intersphinx_registry/lookup.py

```diff
@@ -1,39 +1,61 @@
 import sys
+from typing import Optional
 
 from . import get_intersphinx_mapping
-from . import __version__
-from urllib.parse import urljoin
 
-from typing import Optional
 
-from sphinx.util.inventory import InventoryFile
-from io import BytesIO
+def _check_cli_dependencies():
+    """
+    Check if CLI dependencies are available.
 
-import requests
+    Raises SystemExit with helpful message if dependencies are missing.
+    """
+    missing = []
 
-if len(sys.argv) not in [2, 3]:
-    sys.exit(
-        """Usage: python -m intersphinx_registry.lookup <package>[,package] [search_term]
+    import importlib.util
 
-        Example:
+    missing = []
+    try:
+        import sphinx  # noqa: F401
+        import requests  # noqa: F401
+    except ImportError:
```

Comment:
```suggestion
    except ModuleNotFoundError:
```

`ImportError`s can be anything going wrong, [`ModuleNotFound` (New in 3.6)](https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError) will only be if not installed. You in general never want catch `ImportError`s ever as they indicate something is wrong beyond not being installed.

Reply:
(and that may allow you to not using the find_spec. just two try/except ModuleNotFound)